packages/dd-trace/src/appsec/iast/taint-tracking/rewriter.js

```diff
@@ -114,11 +138,70 @@ function enableRewriter (telemetryVerbosity) {
       }
       shimmer.wrap(Module.prototype, '_compile', compileMethod => getCompileMethodFn(compileMethod))
     }
+
+    enableEsmRewriter(telemetryVerbosity)
   } catch (e) {
     log.error('[ASM] Error enabling TaintTracking Rewriter', e)
   }
 }
 
+function isEsmConfigured () {
+  const hasLoaderArg = !!process.execArgv?.find(
+    arg => arg.includes('--loader') || arg.includes('--experimental-loader')
+  )
+  if (hasLoaderArg) return true
+
+  const nodeOptions = process.env.NODE_OPTIONS
+  const hasLoaderNodeOptions = nodeOptions &&
+    (nodeOptions.includes('--loader') || nodeOptions.includes('--experimental-loader'))
+  if (hasLoaderNodeOptions) return true
```

Comment:
could it be `isFlagPresent('--loader') || isFlagPresent('--experimental-loader')`?

Reply:
Yes, and also I will fix `isFlagPresent` to check it in `NODE_OPTIONS` that is not doing

Reply:
My fault, it is already checking `NODE_OPTIONS` 🤦 
Changed.